lib/dialects/mssql/data-types.js

```diff
@@ -153,6 +153,10 @@ var FLOAT = function() {
 };
 util.inherits(FLOAT, BaseTypes.FLOAT);
 
+BaseTypes.ENUM.prototype.toSql = function() {
+  return 'VARCHAR(10)';
```

Comment:
@janmeier, the function attributeToSQL can't be used to get the SQL for the temp table because it adds extra constraints that are not supported in temp tables.  For example, the enum type will add a check constraint to validate the values, but that isn't supported in temp tables.